orchestra/todos/views.py

```diff
@@ -178,16 +180,23 @@ def patch(self, request, *args, **kwargs):
         return self.put(request, *args, **kwargs)
 
     def perform_update(self, serializer):
-        todo = serializer.save()
-        if isinstance(todo, Todo):
-            old_todo = self.get_object()
-            notify_single_todo_update(
-                self.request.user, old_todo, todo)
+        old_data = copy.deepcopy(serializer.instance)
+        old_todos = list(old_data) if isinstance(
+            old_data, QuerySet) else [old_data]
+        data = serializer.save()
+        todos = data if isinstance(data, list) else [data]
+        for idx, todo in enumerate(todos):
```

Comment:
two thoughts:
1. instead of enumerating `todos` and indexing `old_todos` could you loop through `old_todos` instead?
2. if 1 doesn't work and you need to test the todos in `todos`, you can avoid the enumerate/index by doing `for old_todo, new_todo in zip(old_todos, todos)` which feels a bit cleaner.

Reply:
Avoided the enumerate/index by doing `for old_todo, new_todo in zip(old_todos, todos)`.